tests/ci/report_file_tests.js

```diff
@@ -0,0 +1,103 @@
+var fs = require('fs')
+var App = require('../../lib/ci')
+var TestReporter = require('../../lib/ci/test_reporters/tap_reporter')
+var Config = require('../../lib/config')
+var bd = require('bodydouble')
+var mock = bd.mock
+var stub = bd.stub
+var assert = require('chai').assert
+var expect = require('chai').expect
+var Process = require('did_it_work')
+var exec = require('child_process').exec
+var rimraf = require('rimraf')
+var path = require('path')
+var PassThrough = require('stream').PassThrough
+var ReportFile = require('../../lib/ci/report_file')
+var XUnitReporter = require('../../lib/ci/test_reporters/xunit_reporter')
+var tmp = require('tmp')
+
+describe('report file output', function() {
+  var mainReportDir, reportDir, filename;
+  before(function(done) {
+    tmp.dir(function(err, path) {
+      mainReportDir = path
+      done()
+    })
+  })
+  after(function(done) {
+    rimraf(mainReportDir, function() {
```

Comment:
I tried that and it was still throwing crazy errors on Windows and folders were still hanging around.



On Mar 4, 2015, at 12:10 AM, Johannes W?rbach <notifications@github.com<mailto:notifications@github.com>> wrote:


In tests/ci/report_file_tests.js<https://github.com/airportyh/testem/pull/509#discussion_r25757807>:

> +var path = require('path')
> +var PassThrough = require('stream').PassThrough
> +var ReportFile = require('../../lib/ci/report_file')
> +var XUnitReporter = require('../../lib/ci/test_reporters/xunit_reporter')
> +var tmp = require('tmp')
> +
> +describe('report file output', function() {
> +  var mainReportDir, reportDir, filename;
> +  before(function(done) {
> +    tmp.dir(function(err, path) {
> +      mainReportDir = path
> +      done()
> +    })
> +  })
> +  after(function(done) {
> +    rimraf(mainReportDir, function() {


This could be removed, when tmp.dir({unsafeCleanup: true}, cb) is used in before.

-
Reply to this email directly or view it on GitHub<https://github.com/airportyh/testem/pull/509/files#r25757807>.